html/browsers/the-window-object/self-et-al.window.js

```diff
@@ -0,0 +1,26 @@
+[
+  "frames",
+  "globalThis",
+  "self",
+  "window"
+].forEach(windowProxySelfReference => {
+  test(() => {
+    const frame = document.body.appendChild(document.createElement("iframe")),
+          otherW = frame.contentWindow;
+    assert_equals(otherW[windowProxySelfReference], otherW);
+    frame.remove();
+    assert_equals(otherW[windowProxySelfReference], otherW);
```

Comment:
Would it be interesting or worthwhile to add a test after the pagehide/unload event, just in case some browser has the bright idea to null these things out from a queued task or something?